lib/msal-core/src/AuthResponse.ts

```diff
@@ -4,14 +4,98 @@ import { IdToken } from "./IdToken";
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-export type AuthResponse = {
-    uniqueId: string;
-    tenantId: string;
-    tokenType: string;
-    idToken: IdToken;
-    accessToken: object;
-    scopes: Array<string>;
-    expiresIn: string;
-    account: User;
-    userState: string;
-};
+export class AuthResponse  {
+    private uniqueId: string;
+    private tenantId: string;
+    private tokenType: string;
+    private idToken: IdToken;
+    private accessToken: object;
```

Comment:
should we define a type for accessToken

Reply:
No. We are not supposed to open-up tokens, we do for idToken as we need to read some identifier info like upn etc. So even if we create a type, it will be of the type 'object' - intentional blackbox.

Reply:
why don't we do that? isnt it documentation to the developer what the type is

Reply:
It is by design. Developers or anyone is not supposed to pry open the access_token and base their code on the same, since it is a token and not a set of fixed values, which can change anytime depending on the IDP. Hence we discourage them by not defining a type.

Reply:
makes sense